src/physics/physics.cpp

```diff
@@ -197,6 +202,24 @@ ::post_get_needed_update_flags () const
     return dealii::update_values;
 }
 
+template <int dim, int nstate, typename real>
+template<typename real2>
+real2 PhysicsBase<dim,nstate,real>
+::handle_non_physical_result() const
+{
+    if (this->non_physical_behavior_type == NonPhysicalBehaviorEnum::abort_run) {
+        this->pcout << "ERROR: Non-physical result has been detected. Aborting... " << std::endl << std::flush;
+        std::abort();
+    } else if (this->non_physical_behavior_type == NonPhysicalBehaviorEnum::print_warning) {
+        this->pcout << "WARNING: Non-physical result has been detected at a node." << std::endl;
+    } else if (this->non_physical_behavior_type == NonPhysicalBehaviorEnum::do_nothing) {
+        // do nothing -- assume that the test or iterative solver can handle this.
+    }
```

Comment:
maybe add something like `const std::string non_physical_qty_name` as an argument to this function so that it prints what quantity is nonphysical like it did before in Euler

Reply:
Good suggestion. I've made a change which prints an error message.